packages/utils/telemetry-utils/src/eventEmitterWithErrorHandling.ts

```diff
@@ -12,20 +12,11 @@ import { TypedEventEmitter, EventEmitterEventType } from "@fluidframework/common
  */
 export class EventEmitterWithErrorHandling<TEvent extends IEvent = IEvent> extends TypedEventEmitter<TEvent> {
     public emit(event: EventEmitterEventType, ...args: any[]): boolean {
-        if (event === "error") {
```

Comment:
Removal of this code (I believe) will result in exception thrown through our stack (if there is no "error" handler setup).
I do not think allowing exceptions through our stack is a good thing. I think much better would be to be explicit, i.e. require a callback from user of this class to call in such case and decide what to do. Most of our code should play safe and close container with critical error, as both letting exception through and eating it are two bad choices (former might be slightly better because in most cases it will cause container close, but there are no guarantees here)

Reply:
> Removal of this code (I believe) will result in exception thrown through our stack

It's already the case, that's why I'm making this change.

Reply:
Vlad and I discussed on the phone, the plan is to update this class to take an explicit callback to be invoked if a event listener throws, rather than leaning on the error event.